src/main.rs

```diff
@@ -203,7 +201,7 @@ fn main() {
 fn snapshot(pid: pid_t) -> Result<(), Error> {
     let getter = initialize::initialize(pid)?;
     let trace = getter.get_trace()?;
-    for x in trace.iter().rev() {
+    for x in trace.trace.iter().rev() {
```

Comment:
`trace.trace` feels like there is an accidental bit of too much indirection.

Reply:
yep agreed